react-front-end/__tests__/tsrc/search/AdvancedSearch.test.tsx

```diff
@@ -285,11 +300,67 @@ describe("Rendering of wizard", () => {
     mockGetAdvancedSearchByUuid.mockResolvedValue(advancedSearchDefinition);
     const { container } = await renderAdvancedSearchPage();
 
-    // Collect all default values.
-    const defaultLabelsAndValues = mockedControls.map(([c, controlValue]) =>
-      getControlValue(container, Array.from(controlValue.keys()), c.controlType)
+    // Collect _all_ values.
+    const defaultLabelsAndValues = buildMockedControls().map(
+      ([c, controlValue]) =>
+        getControlValue(
+          container,
+          Array.from(controlValue.keys()),
+          c.controlType
+        )
     );
 
+    // Make sure the current values match the default
     expect(defaultLabelsAndValues).toEqual(mockedLabelsAndValues);
+
+    // Clear all input value
+    await clickClearButton(container);
+
+    const afterClearLabelsAndValues: WizardControlLabelValue[] = pipe(
+      buildMockedControls(),
+      A.map<MockedControlValue, WizardControlLabelValue>(([c, controlValue]) =>
+        pipe(
+          getControlValue(
+            container,
+            Array.from(controlValue.keys()),
+            c.controlType
+          ),
+          O.fromNullable,
+          O.getOrElse(() => new Map())
+        )
+      )
+    );
+
+    const isNotEmptyMap = (m: Map<unknown, unknown>): boolean => {
+      return !M.isEmpty(m);
+    };
+
+    // filter all empty value
+    const filteredLabelsAndValues = pipe(
+      afterClearLabelsAndValues,
+      // filter {}
+      A.filter(isNotEmptyMap),
+      // filter "" and []
+      A.filter(
+        flow(
+          M.filter((mapValue) => !A.isEmpty(mapValue as string[])),
+          isNotEmptyMap
+        )
+      ),
+      // filter ["", ""]
+      A.filter(
+        flow(
+          M.filter((m) =>
+            flow(
+              A.filter((arrayItem) => arrayItem !== ""),
+              A.isNonEmpty
+            )(m as string[])
+          ),
+          isNotEmptyMap
+        )
```

Comment:
And let's continue the theme:

```suggestion
        flow(
          M.filter(flow(
              A.filter(not(S.isEmpty)),
              A.isNonEmpty
            )
          ),
          not(M.isEmpty)
        )
```

Not also here I've dropped your type coercion (i.e. `m as string[]`). You don't want to be doing that at all, most of the type you can solve this by explicitly specifying the types for the generics. So if you take the above approach but then have type issues, look at the generic type specifiers rather than adding the coercion back in.

Reply:
Still got errors by explicitly specifying the types for generics. I tried to use ternary to avoid the error. Not sure whether it's a good solution? See the latest commit.